src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -131,149 +131,143 @@ writeSys wt d k v = do
         Namespaces -> convNamespaceName
         Pacts -> convPactId
         UserTables _ -> error "impossible"
-    write key bh version txid db =
+    write key bh txid db =
         let f = case wt of
                     Insert -> backendWriteInsert
                     Update -> backendWriteUpdate
                     Write -> backendWriteWrite
-        in f key (domainTableName d) bh version txid v db
+        in f key (domainTableName d) bh txid v db
 
 backendWriteInsert
     :: ToJSON v
     => Utf8
     -> Utf8
     -> BlockHeight
-    -> ReorgVersion
     -> TxId
     -> v
     -> Database
     -> IO ()
-backendWriteInsert key tn bh version txid v db =
+backendWriteInsert key tn bh txid v db = do
     exec' db q [ SText key
                , SInt (fromIntegral bh)
-               , SInt (fromIntegral version)
                , SInt (fromIntegral txid)
                , SBlob (toStrict (Data.Aeson.encode v))]
+    exec' db mutq [SText tn, SInt (fromIntegral bh)]
   where
+    mutq = "INSERT OR IGNORE INTO VersionedTableMutation VALUES (?,?);"
     q = mconcat [ "INSERT INTO ["
                 , tn
-                , "] ('rowkey','blockheight','version','txid','rowdata') \
-                  \ VALUES (?,?,?,?,?);"
+                , "] ('rowkey','blockheight','txid','rowdata') \
+                  \ VALUES (?,?,?,?);"
                 ]
 
-
 backendWriteUpdate
     :: ToJSON v
     => Utf8
     -> Utf8
     -> BlockHeight
-    -> ReorgVersion
     -> TxId
     -> v
     -> Database
     -> IO ()
-backendWriteUpdate key tn bh version txid v db =
+backendWriteUpdate key tn bh txid v db = do
   exec' db q [ SBlob (toStrict (Data.Aeson.encode v))
              , SText key
              , SInt (fromIntegral bh)
-             , SInt (fromIntegral version)
              , SInt (fromIntegral txid)
              ]
+  exec' db mutq [SText tn, SInt (fromIntegral bh)]
   where
+    mutq = "INSERT OR IGNORE INTO VersionedTableMutation VALUES (?,?);"
     q = mconcat [ "UPDATE ["
                 , tn
                 , "] SET rowdata = ? WHERE rowkey = ? AND blockheight = ? \
-                  \ AND version =  ? AND txid = ? ;"
+                  \ AND txid = ? ;"
```

Comment:
Wait a minute -- shouldn't update produce a new row rather than being a SET operation? If we mutate data tied to an old txid then we are going to break rewind. This is a bug on the original branch too :(

Reply:
This was resolved on the branch `feat/pact-replay`. I'll make sure to merge in the latest changes from that branch.